src/components/Tables/TableFilterAndExtraColumnsBar.tsx

```diff
@@ -64,12 +63,36 @@ export const filtersToComponents = <Data,>(
     });
 };
 
-const TableFilterAndExtraColumnsBar = <Data,>(props: Props<Data>): React.ReactElement => {
+const TableFilterAndExtraColumnsBar = <Data, Filter extends CoreFilter>(
+    props: Props<Data, Filter>
+): React.ReactElement => {
+    const handleClear = (): void => {
+        if (props.setFilters && props.filters) {
+            props.setFilters(
+                props.filters?.map((filter) => ({
+                    ...filter,
+                    state: filter.initialState,
+                }))
+            );
+        }
+        if (props.setAdditionalFilters && props.additionalFilters) {
+            props.setAdditionalFilters(
+                props.additionalFilters?.map((filter) => ({
+                    ...filter,
+                    state: filter.initialState,
+                }))
+            );
+        }
+    };
+
     const [showMoreFiltersAndHeaders, setShowMoreFiltersAndHeaders] = useState(false);
 
-    const hasPrimaryFilters = props.filters.length !== 0;
+    const hasPrimaryFilters = props.filters && props.filters?.length !== 0 && props.setFilters;
 
-    const hasAdditionalFilters = props.additionalFilters.length !== 0;
+    const hasAdditionalFilters =
+        props.additionalFilters &&
+        props.additionalFilters?.length !== 0 &&
```

Comment:
can be simplified i think
```suggestion
        props.additionalFilters?.length > 0 &&
```

Reply:
if I use `!== 0` it works but if I use `> 0` I get an error saying props.additionalFilters.length might be undefined, so have used the former